src/App.tsx

```diff
@@ -16,104 +11,90 @@ function MidiToNotes (midiNotes: {[key: number]: Note}) : Notes {
+    pressedKeys: {},
+  })
 
-    // Used to detect a potential race condition between WebMidi and React while mounting the component
-    this._mounted = false;
-
-    WebMidi.enable((err) => {
+  useEffect(() => {
+    WebMidi.enable((err) => { 
       if (err) {
         console.log("WebMidi could not be enabled.", err);
-        if (this._mounted) {
-          this.setState(() => { return { webMidiStatus: "error" }; });
-        } else {
-          this.state = Object.assign(this.state, { webMidiStatus: "error" });
-        }
+        setState({
+          ...state,
+          webMidiStatus: "error"
+        });
       }
       else {
         console.log("WebMidi enabled!");
-        if (this._mounted) {
-          this.setState(() => { return { webMidiStatus: "initialized" }; });
-        } else {
-          this.state = Object.assign(this.state, { webMidiStatus: "initialized" });
-        }
-        // this.setState(() => { return {webMidiStatus: "initialized"}; } );
-        // this.setState({webMidiStatus: "initialized"});
+        setState({
+          ...state,
+          webMidiStatus: "initialized"
+        });
       }
-    });
-    this.state = {
-      access: navigator.requestMIDIAccess(),
-      webMidiStatus: "initializing",
-      pressedKeys: {},
-    };
-  }
+    })
+  }, [state])
 
-  componentDidMount(): void {
-    this._mounted = true;
-  }
 
-  onSelectedInput(e: React.ChangeEvent<HTMLSelectElement>): void {
+  const onSelectedInput = (e: React.ChangeEvent<HTMLSelectElement>): void => {
     // Cleanup old input
-    if (this.state.selected_input) {
-      const oldInput = WebMidi.getInputById(this.state.selected_input);
+    if (state.selected_input) {
+      const oldInput = WebMidi.getInputById(state.selected_input);
       if (oldInput) {
         oldInput.removeListener('noteon');
         oldInput.removeListener('noteoff');
       }
     }
 
     const inputId = e.target.value;
-    this.setState({ selected_input: inputId });
+    setState({ 
+      ...state,
+      selected_input: inputId 
+    });
     const input = WebMidi.getInputById(inputId);
 
-
     // Listen for a 'note on' message on all channels
     if (input) {
       input.addListener('noteon', "all",
         (e) => {
-          let pressedKeys = Object.assign(this.state.pressedKeys, {[e.note.number]: e.note.name as Note});
-          this.setState({pressedKeys});
+          let pressedKeys = Object.assign(state.pressedKeys, {[e.note.number]: e.note.name as Note});
+          setState({
+            ...state,
+            pressedKeys: pressedKeys});
         }
       );
       input.addListener('noteoff', "all",
          (e) => {
-          const pressedKeys = Object.assign(this.state.pressedKeys);
+          const pressedKeys = Object.assign(state.pressedKeys);
           delete pressedKeys[e.note.number];
-          this.setState({pressedKeys});
+          setState({
+            ...state,
+            pressedKeys: pressedKeys});
```

Comment:
We could again use an updater. This would mean to delete the two lines as marked above and here it would look something like this (up to indentation):

```suggestion
          setState((prevState) => {
              const pressedKeys = Object.assign(prevState.pressedKeys);
              delete pressedKeys[e.note.number];
              return {
                  ...prevState,
                  pressedKeys: pressedKeys}
             });
```